aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MySecondController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+
+namespace performance_best_practices.Controllers
+{
+    // Requires review, Original
+    //  var form = HttpRequest.Form;
+    [Route("api/[controller]")]
+    [ApiController]
+    #region snippet1
+
+    public class BadReadController : Controller
+    {
+        [HttpPost("/form-body")]
+        public IActionResult Post()
+        {
+            var form =  HttpContext.Request.Form;
```

Comment:
This is misleading to show in a controller, MVC usually reads the form for you async during model binding.

Reply:
@Tratcher  this comes from https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AspNetCoreGuidance.md#prefer-using-httprequestreadasformasync-over-httprequestform
I'm not allowed to change code so you'll have to provide new code.

Reply:
We can change it to middleware but the point is to show both